### ISSUE TITLE
[css-transforms-1] The computed value of `transform` if the transformed element is display:none

### ISSUE THREAD

**Issue by BorisChiou** [w3c/csswg-drafts#9121]
Just curious about where we define the computed values of `transform` property for `display:none` element (or in the `display:none` subtree).

Per this [WPT PR#36380](https://github.com/web-platform-tests/wpt/pull/36308), it expects the serialization of the computed value of `transform` property is `none` if we set this element `display:none` (or in the `display:none` subtree). However, I cannot find the specific words in the spec (e.g. [css-transforms-1](https://drafts.csswg.org/css-transforms/) and [css-display-3](https://drafts.csswg.org/css-display/)) to mention about this behavior.

Should we add this into the spec to specify this behavior because it seems this is only for `transform` property? How about `translate`/`rotate`/`scale` properties? Should they be consistent?

**Comment by Loirooriol** [comment]
The test is actually checking the resolved value. For properties that have a special resolved value, it tends to be the computed value when in `display: none`. Making the computed value become `none` in `display: none` seems strange to me.

**Comment by nt1m** [comment]
WebKit originally changed to stop return `none` when the display value was `none` in https://github.com/WebKit/WebKit/commit/ae44d0ca68fa6c38c22fad2efbc2dbdc45a4f581 , but that part was reverted in https://github.com/WebKit/WebKit/commit/3a3a97e3b21468b4283bdbf56d66da3299715dbc due to breaking a site (although that site can probably be fixed on our end, I would be cautious about potential web compat impact here).

**Comment by dbaron** [comment]
We had some Chromium discussion of this in https://chromium-review.googlesource.com/c/chromium/src/+/4878164?tab=comments , but really the conclusion there was to fix the test to accept multiple possible behaviors (as discussed above in https://github.com/w3c/csswg-drafts/issues/9121#issuecomment-1683001516) until this issue is resolved.

As far as sensible behavior goes:
* I think it's strange to return a `matrix()` for elements that are `display: none` because doing so, for some cases (resolving percentages) requires making up a size of box to resolve those percentages against.
* I also think it's strange to return `none` (as the resolved value) when the computed value isn't `none`, with the exception of cases where `transform` doesn't apply (like `display: inline`) where I think returning `none` does make sense.
* It's probably bad to return something closer to the computed value (a list of functions) because it's bad for calling code to get something other than a `matrix()` for edge cases when the calling code expects either `matrix()` or `none`.

So, basically, I think all three possible behaviors that I've thought of are bad.

It sounds like we also need to worry a good bit about compatibility here, which may mean that we should just pick the bad thing that's web-compatible instead of the bad things that aren't web-compatible.

**Comment by dbaron** [comment]
(Of the three options I wrote, I think the `matrix()` one seems least bad from first principles, though I didn't reread the compatibility information.)

**Comment by kbabbitt** [comment]
As I understand it, the one bit of web compatibility data we have is that WebKit regressed a site when they changed away from returning `none`. I also did some digging and found that an older version of the spec [1] defined "transformable element" in a way that excludes `display:none` elements, in which case `transform` would not apply to `display:none` elements, which I think makes it reasonable for the computed value to be `none`.

So right now I'd favor the `none` option but would reconsider if more compatibility data surfaced.

[1] https://www.w3.org/TR/2017/WD-css-transforms-1-20171130/#terminology
